File: UniEatsBackEnd/Models/Reservation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace UniEatsBackEnd.Models
{
    public class Reservation
    {
        [Key]
        public int ReservationId { get; set; }

        [Required]
        public int UserId { get; set; }

        [Required]
        public DateTime ReservationDate { get; set; } = DateTime.Now;

        [Required]
        [Range(1, int.MaxValue, ErrorMessage = "Number of people must be greater than 0.")]
        public int NumberOfPeople { get; set; }

        [Required]
        [RegularExpression("(Confirmed|Canceled)", ErrorMessage = "Invalid status.")]
        public string Status { get; set; } = "Confirmed";

        [Required]
        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: UniEatsFrontEnd/UniEatsFrontEnd/src/models/reservation.ts
import { User } from './user';

export interface Reservation {
  reservationId: number; // Unique identifier for the reservation
  userId: number; // ID of the user who made the reservation
  reservationDate?: Date; // Defaults to the current date
  numberOfPeople: number; // Number of people for the reservation, must be > 0
  status: 'Confirmed' | 'Canceled'; // Defaults to 'Confirmed'
  tableNumber: number; // Table number, must be > 0
  createdAt?: Date; // Defaults to the current date
  user?: User | null; // Optional navigation property for the associated user
}
